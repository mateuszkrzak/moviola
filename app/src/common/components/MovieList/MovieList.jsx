import React from 'react';
import PropTypes from 'prop-types';

import './MovieList.scss';

import MovieElement from './MovieElement/MovieElement';

export default class MovieList extends React.Component {
  static defaultProps = {
    movies: [],
  };

  static propTypes = {
    movies: PropTypes.arrayOf(PropTypes.object),
  };

  render() {
    if (!this.props.movies.length) {
      return <div styleName="no-results">No films found</div>;
    }

    return (
      <div styleName="list">
        {this.props.movies.map(movie => <MovieElement key={movie.id} movie={movie} />)}
      </div>
    );
  }
}
