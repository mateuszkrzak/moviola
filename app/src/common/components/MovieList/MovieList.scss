.list {
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  grid-gap: 20px;
  margin-bottom: 2em;
}

.no-results {
  font-size: 44px;
  margin-top: 5em;
  text-align: center;
}
