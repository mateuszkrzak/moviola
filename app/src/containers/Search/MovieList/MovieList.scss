.list {
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  grid-gap: 20px;
}

.no-results {
  font-size: 44px;
  margin-top: 5em;
  text-align: center;
}
