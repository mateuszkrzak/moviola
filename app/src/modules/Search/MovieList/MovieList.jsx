import React from 'react';
import CSSModules from 'react-css-modules';
import PropTypes from 'prop-types';

import styles from './MovieList.scss';

import MovieElement from './MovieElement/MovieElement';

class MovieList extends React.Component {
  static defaultProps = {
    movies: [],
  };

  render() {
    return (
      <div styleName="list">
        {this.props.movies.map(movie => <MovieElement key={movie.id} movie={movie} />)}
      </div>
    );
  }
}

MovieList.propTypes = {
  movies: PropTypes.arrayOf(PropTypes.object),
};

export default CSSModules(MovieList, styles);
