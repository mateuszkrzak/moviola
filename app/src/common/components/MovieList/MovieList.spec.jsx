import { shallow } from 'enzyme';
import React from 'react';
import MovieList from './MovieList';

test('MovieList component renders "No films found" if passed movies array is empty', () => {
  const mockMovies = [];
  const wrapper = shallow(<MovieList movies={mockMovies} />);
  expect(wrapper).toMatchSnapshot();
});

test('MovieList component renders MovieElement components', () => {
  const mockMovies = [{ id: 1 }];
  const wrapper = shallow(<MovieList movies={mockMovies} />);
  expect(wrapper).toMatchSnapshot();
});
