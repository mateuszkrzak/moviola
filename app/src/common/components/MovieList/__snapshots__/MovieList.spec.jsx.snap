// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieList component renders "No films found" if passed movies array is empty 1`] = `
<div
  styleName="no-results"
>
  No films found
</div>
`;

exports[`MovieList component renders MovieElement components 1`] = `
<div
  styleName="list"
>
  <MovieElement
    genres={Array []}
    id={1}
    key="1"
    poster_path=""
    release_date=""
    title=""
  />
</div>
`;
